hiku/graph.py

```diff
@@ -131,6 +133,10 @@ def __repr__(self):
     def options_map(self):
         return OrderedDict((op.name, op) for op in self.options)
 
+    @cached_property
+    def directives_map(self):
```

Comment:
According to federation spec, https://www.apollographql.com/docs/federation/federation-spec/#key, there can be multiple same directives, so directives_map is definitely a wrong decision.

Reply:
And `_FieldSet` type can have multiple values, even nested 🤯 I wonder why they don't use `String` type here with ability to specify several directives

Reply:
I haven't implemented multiple and nested values. For now just single field like `id` or `cartId`

Reply:
But I assume, since _FieldSet is a selection set, we have to parse fields string into selection set

Reply:
> I haven't implemented multiple and nested values. For now just single field like `id` or `cartId`

Probably it would be enough to start with.

> But I assume, since _FieldSet is a selection set, we have to parse fields string into selection set

I was thinking that `@key` directive is more for the federation gateway service, not for us. Maybe I'm missing something, I'm not finished reviewing this PR.

Reply:
@key directive is a way of saying which field in type has to be a primary key. In prom, we have Order and Cart types, and since these types are in different services, we have to use @key to define primary keys.

Reply:
here is an example how we can extend some types from one service in another. Here we are extending `Order` node with `cart` field, and declaring `Product` stub type, so we can add `product` field into `CartItem` type in cart service
```

extend_nodes = [
    Node('Order', [
        Field('cartId', Integer, ids_resolver,
              directives=[External()]),
        Link('cart', TypeRef['Cart'], direct_link, requires='cartId'),
    ], directives=[Key('cartId'), Extends()]),
    Node('Product', [
        Field('id', Integer, ids_resolver, directives=[External()]),
    ], directives=[Key('id'), Extends()]),
]

```